packages/gatsby-recipes/src/providers/npm/package.js

```diff
@@ -60,34 +74,59 @@ const executeInstalls = async root => {
     packageManager: PACKAGE_MANGER,
   })
 
-  try {
-    await execa(PACKAGE_MANGER, commands, {
-      cwd: root,
-    })
-  } catch (e) {
-    // A package failed so call the rejects
-    return packagesToInstall.forEach(p => {
-      p.outsideReject(
-        JSON.stringify({
-          message: e.shortMessage,
-          installationError: `Could not install package`,
+  const release = await lock(`package.json`)
+  if (isJest) {
```

Comment:
This isn't a great pattern for testing. For one, we should test production code, not a separate path. And two, any test-specific logic should live in testing files rather than the main codebase.

Reply:
👍 We should mock execa and execute the jestCode that you wrote here so the flow inside this file isn't interrupted by an environment check.